qiskit/primitives/base/base_sampler.py

```diff
@@ -192,3 +196,14 @@ def parameters(self) -> tuple[ParameterView, ...]:
             List of the parameters in each quantum circuit.
         """
         return tuple(self._parameters)
+
+
+def _has_measure(circuit: QuantumCircuit) -> bool:
+    for instruction in circuit:
```

Comment:
Since measurements are typically at the end of the circuit, in standard cases this check should return faster if you iterate over the reverse of the circuit data.

Reply:
I see. `for instruction in circuit[::-1]:` makes the check faster for standard cases.

Reply:
Thanks. I will follow your great suggestion.